### ISSUE TITLE
Animate the Empty State

### ISSUE THREAD

**Issue by grgia** [Expensify/App#19954]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

We have some lottie animations, let's use them + add a gradient and animate the empty state. 

The background animation will need to work for both themes.

<img src="https://github.com/Expensify/App/assets/38015950/4696e57e-f674-4810-9f7c-b0d0b2b6c3b3">

<img src="https://github.com/Expensify/App/assets/38015950/546ceae1-2a92-40f7-9cfd-ef35a76f92d0" width="300">

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0125742c76895ef142</li>
        <li>Upwork Job ID: 1664270385032765440</li>
        <li>Last Price Increase: 2023-06-01</li>
    </ul>
</details>

**Comment by shawnborton** [comment]
Will share a Lottie file as soon as I have it!

**Comment by grgia** [comment]
Link to Lottie files: https://expensify.slack.com/archives/C03KN50J0PM/p1685655362569899?thread_ts=1685627642.791289&cid=C03KN50J0PM

**Comment by grgia** [comment]
@shawnborton I'm wondering if we should just add the gradient to the images themselves rather than masking with a gradient- is there a reason to use a flat color here?

**Comment by grgia** [comment]
Here's an example with a gradient mask on the left side, so it's definitely an option but I might need some specifics for the mask if we choose this route:
<img width="788" alt="image" src="https://github.com/Expensify/App/assets/38015950/5eff6e4f-3561-4690-b317-728b13d44c46">

**Comment by shawnborton** [comment]
We can do whatever you think is easiest - but looks like the mask might work well for us? My thinking for the mask is that it would be gradient that looks something like this:
<img width="612" alt="image" src="https://github.com/Expensify/App/assets/2319350/2593e98a-bbd4-4485-a5b1-cc30f5bbaaf8">

```background: linear-gradient(0deg, #061B09 0%, rgba(6, 27, 9, 0) 100%);```

**Comment by grgia** [comment]
@shawnborton for that mask example, what is the height of the mask over the image? Is it 100% height and width?

**Comment by shawnborton** [comment]
Yup exactly!

**Comment by grgia** [comment]
still working on this

**Comment by thienlnam** [comment]
We already have an issue to animate the empty state FYI - linking here so we don't do duplicate work https://github.com/Expensify/App/issues/16366

**Comment by grgia** [comment]
@thienlnam this is just adding the lottie animations- the linked issue is for tilt screen animations- I don't think it's dupe

**Comment by roryabraham** [comment]
@grgia agree this is not a dupe, but maybe we should clarify this issue title like "Add empty state illustrations to thread reports".

**Comment by grgia** [comment]
not overdue

**Comment by grgia** [comment]
not overdue, paused for rebrandify

**Comment by shawnborton** [comment]
Georgia is OOO this week, we're still holding.

**Comment by shawnborton** [comment]
I think we're still holding, but if it's going to take a while to get the animations in place, I'd be down to just mimic the static style we currently use as a placeholder. Thoughts?

**Comment by grgia** [comment]
> I think we're still holding, but if it's going to take a while to get the animations in place, I'd be down to just mimic the static style we currently use as a placeholder. Thoughts?

That's an easy fix. I can likely work with @roryabraham to debug the lottie resizing issues I was having when we're at offshore

**Comment by grgia** [comment]
@roryabraham and I were talking and the reason that I couldn't get the resizing to work before was due to the lottie library missing web features. There's been an issue opened that SWM is working on to add web support so I think we should HOLD on that.

**Comment by grgia** [comment]
@shawnborton in the meantime, I can create an issue to replace the background image with a new still version?

**Comment by shawnborton** [comment]
That would be great Georgia! We can just use the same still BG version that we currently use for empty state everywhere, we just need to implement it on threads/expenses/reports/etc

**Comment by grgia** [comment]
Hey @roryabraham do you have the issue link for fixing Lottie for web that SWM is working on?

**Comment by roryabraham** [comment]
https://github.com/Expensify/App/issues/26871

**Comment by grgia** [comment]
on hold

**Comment by shawnborton** [comment]
Not overdue, still holding

**Comment by grgia** [comment]
still hold

**Comment by roryabraham** [comment]
https://github.com/Expensify/App/issues/26871 is prettymuch done, taking this off hold and moving back to weekly.

**Comment by grgia** [comment]
not overdue. hey @chrispader how well would lottie animations fit into the illustration provider if we plan on making this change?

**Comment by chrispader** [comment]
I think we can just use it similarly, since would essentially just import different lottie files or adapt the color somehow:

Therefore we could just move this code to the provider (for dynamic lottie animations)

```
const DotLottieAnimations: Record<string, DotLottieAnimation> = {
    ExpensifyLounge: {
        file: require('@assets/animations/ExpensifyLounge.lottie'),
        w: 1920,
        h: 1080,
    },
...
```

**Comment by grgia** [comment]
@chrispader  sweet I think we can make this change as a follow up after we go live 🎉

**Comment by shawnborton** [comment]
Still not a priority it seems, so not overdue.

**Comment by shawnborton** [comment]
Still not a priority. @grgia do you still think we should do this one?

**Comment by shawnborton** [comment]
Not overdue, but not a priority.

**Comment by grgia** [comment]
Checking see if anyone from callstack wants to take this https://expensify.slack.com/archives/C03UK30EA1Z/p1711454705303899

**Comment by grgia** [comment]
Posted in SWM https://expensify.slack.com/archives/C04878MDF34/p1711560191194859

Otherwise, going to make this [external from this point](https://expensify.slack.com/archives/CC7NECV4L/p1711559802930789?thread_ts=1709812877.445629&cid=CC7NECV4L)

**Comment by blazejkustra** [comment]
Hey! @SzymczakJ is interested, he'll post a comment today morning 😄

**Comment by SzymczakJ** [comment]
Hey! I’m Jakub Szymczak from Software Mansion, an expert agency, and I’d like to work on this issue!

**Comment by grgia** [comment]
Try these! 
[ExpensifyLottieFiles_060123_2 (1).zip](https://github.com/Expensify/App/files/14835076/ExpensifyLottieFiles_060123_2.1.zip)

**Comment by grgia** [comment]
https://expensify.slack.com/archives/C03KN50J0PM/p1685627642791289

[ExpensifyLottieFiles_060123 (1).zip](https://github.com/Expensify/App/files/14853246/ExpensifyLottieFiles_060123.1.zip)


dark mode assets / original slack thread for future ref!

**Comment by SzymczakJ** [comment]
I have a design question: 
Right now we have an image in empty state background and this image is repeated when image width is smaller than window width which results in this:
<img width="1920" alt="Screenshot 2024-04-03 at 17 01 56" src="https://github.com/Expensify/App/assets/88395093/7973290a-37d8-416b-aaee-8784318cfb83">

Such 'repeat' feature won't be done easily since Lottie doesn't support this. Can we stick to the single animation in the center? It will simplify the code by a lot and personally I think it looks better:
<img width="1913" alt="Screenshot 2024-04-03 at 17 02 56" src="https://github.com/Expensify/App/assets/88395093/f602b40e-ef79-41c8-8dc2-2aacdb45606c">
@shawnborton

**Comment by shawnborton** [comment]
> Can we stick to the single animation in the center? It will simplify the code by a lot and personally I think it looks better:

I think we could, but perhaps we should make the image way bigger. Ideally we want this to feel as similar to the existing empty state as possible.

**Comment by SzymczakJ** [comment]
Making image bigger when display width is bigger than image is definitely a good idea! How much bigger should the image be? Right now image height for a big screen is 450px. **My proposition is: when the image width becomes smaller than display width, then we could enlarge it so the height is for example 800px(width will be scaled accordingly).** This is how this looks like when display width is almost 3000px
<img width="1919" alt="2880animation" src="https://github.com/Expensify/App/assets/88395093/72514934-5886-4d2c-ac68-27d524286c09">
What do you think @shawnborton @grgia?

**Comment by SzymczakJ** [comment]
One more thing that would make animation look better and make it more similar to the existing background:
Animation has a lot of free space at the top, compared to the existing background.
Existing background:
<img width="1300" alt="backroundimg" src="https://github.com/Expensify/App/assets/88395093/5d646178-8e8c-4546-b949-03c8169d7088">
New animation:
<img width="1295" alt="backgroundani" src="https://github.com/Expensify/App/assets/88395093/d3876107-8e8b-499c-9be6-44949055a7ee">
As we can see there is a lot of unused empty space. Can we tweak this lottie animation so the empty space is minimalized, as it is in the current background?

**Comment by shawnborton** [comment]
Hmm yeah, good shout there. I bet we can update the animation to better match the existing dimensions of the empty state. Can you let me know what the existing empty state dimensions are? Perhaps even attach the svg here too if you can - thanks!

**Comment by SzymczakJ** [comment]
Current state(original image dimensions are 3896 × 1048):
[images.zip](https://github.com/Expensify/App/files/14871882/images.zip)

New animations(original image dimensions are 3750 × 1120):
[animations.zip](https://github.com/Expensify/App/files/14871909/animations.zip)

**Comment by shawnborton** [comment]
Hmm those are valid points. This is really just design polish at this point and the current empty state seems to be working just fine. I would be fine closing this one out so we can focus on higher priorities.

**Comment by shawnborton** [comment]
Going to close this then - sorry for the false alarm @SzymczakJ

**Comment by shawnborton** [comment]
Yeah, I think there are a couple of things that we need to consider here:
1. [We are changing the way the empty state works](https://github.com/Expensify/App/pull/38449), in that we are going to make it take up less scrolling space and feel more like a background element. It doesn't change much, but I think we should wait until that is finished before doing anything else. 
2. We'd definitely need to have this animation recreated in the new dimensions/size, and I'm not sure if it's worth the effort/cost
3. Have we investigated how this might impact performance? This would put a lottie animation on virtually every single chat view in the app. So I think I am a little cautious to move forward after thinking more on that. Thoughts?

**Comment by SzymczakJ** [comment]
> 3. Have we investigated how this might impact performance? This would put a lottie animation on virtually every single chat view in the app. So I think I am a little cautious to move forward after thinking more on that.
I probably should have started with checking the performance. I've been developing this on web and it worked smoothly, but now I've checked it on a real iOS device and that's how it looks like:

https://github.com/Expensify/App/assets/88395093/3e563240-0715-4438-8797-6e150e1c8b95

There is a visible drop of frames when scrolling through chat with having an animation in the background. This animation consists of many elements, so the drop of frames is normal. After talking to some animation experts in SWM and trying Lottie optimisation tools I don't think we can speed up this animation in any way apart from making the animation simpler.
**TL;DR performance on real devices is bad, I think we have to make this animation simpler(not having so many animated elements, not being so complex etc.)**

**Comment by grgia** [comment]
Hmm, given the performance impact/needing to update assets, maybe @shawnborton's right that it's best to table this issue for now. Do you agree?

**Comment by shawnborton** [comment]
Yeah, I'm thinking we just table this.
